metrique/tests/ui/fail/bad_field_attrs.rs

```diff
@@ -34,11 +34,27 @@ struct MetricsWithInvalidUnit {
 
     #[metrics(prefix = "foo")]
     prefix_no_flatten: usize,
+
+    #[metrics(flatten, prefix = "foo@")]
+    prefix_bad_character: usize,
+
+    #[metrics(flatten, prefix = "foo", explicit_prefix = "bar")]
+    prefix_and_explicit: SubMetric,
 }
 
 #[metrics(rename_all = "snake_case")]
 struct SubMetric {
     a: usize,
 }
 
+#[metrics(prefix = "foo", explicit_prefix = "foo")]
+struct PrefixAndExplicit {
+    a: usize,
+}
+
+#[metrics(prefix = "foo@")]
```

Comment:
just for the purposes of integration testing, we should probably use `:` that is the field most commonly used in the wild right now